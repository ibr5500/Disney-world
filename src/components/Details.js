import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';

const Details = () => {
  const character = useParams();
  const details = useSelector((state) => state.characters);
  const charDetails = details.filter((char) => +char.id === +character.id);

  return (
    <div className="details-container">
      {charDetails.map((char, index) => (
        <div className="details" key={index.legth * 5 + 1}>
          <div className="detail-img">
            <img src={char.image} alt={char.name} />
          </div>
          <div className="name">
            <h4>
              <span>Name:</span>
              {'  '}
              {(char.name[0] === '.' || char.name[0] === "'")
                ? char.name.replace(/[.']/g, '')
                : char.name}
            </h4>
          </div>
          <ul className="movies">
            <h4>Movies: </h4>
            {(!char.movies.length) ? <p>This character did not participate in Movies.</p>
              : char.movies.map((movie) => (
                <li key={index.legth * 5 + 1}>
                  {movie}
                  .
                </li>
              ))}
          </ul>
          <ul>
            <h4>Tv Shows: </h4>
            {(!char.shows.length) ? <p>This character did not participate in TV Shows.</p>
              : char.shows.map((show) => <li key={index.legth * 5 + 1}>{show}</li>)}
          </ul>
        </div>
      ))}
    </div>
  );
};

export default Details;
