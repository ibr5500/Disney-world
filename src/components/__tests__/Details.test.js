import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Details from '../Details';

describe('Test for Details page', () => {
  it('Render the Details', () => {
    const tree = renderer.create(<Provider store={store}><Details /></Provider>);
    expect(tree).toMatchSnapshot();
  });
});