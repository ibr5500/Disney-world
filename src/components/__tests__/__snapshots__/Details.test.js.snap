// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Details page Render the Details 1`] = `
<div
  className="details-container"
/>
`;
